lib/pos_epoch.go

```diff
@@ -84,7 +98,16 @@ func (bav *UtxoView) GetCurrentEpochEntry() (*EpochEntry, error) {
 
 	// If still not found, return the GenesisEpochEntry. This will be the
 	// case prior to the first execution of the OnEpochCompleteHook.
-	return &EpochEntry{EpochNumber: 0, FinalBlockHeight: math.MaxUint64}, nil
+	//
+	// TODO: Should FinalBlockHeight be ProofOfStake1StateSetupBlockHeight for epoch 0?
+	// The ProofOfStake1StateSetupBlockHeight fork height is exactly when epoch 0 ends.
+	// Epoch 1 begins at the next height.
```

Comment:
We currently don't read the FinalBlockHeight for epoch 0 so this shouldn't matter. That said, `ProofOfStake1StateSetupBlockHeight` is technically the correct value to return and IMO will avoid confusion for the long run. I'm leaving this as a TODO here since it's not clear to me what the motivation was to return `math.MaxUint64` and if it has some special purpose

Reply:
I think we just wanted a way to detect that we were at the genesis. So it's kindof like a "marker" value. That said, I agree the ProofOfStake1StateSetupBlockHeight is a better thing to return. Feel free to change it. Just make sure you also change what calls it so that they're properly picking up the market value.